root_VS2012/programs/C#/Frameworks/Infrastructure/Public/Db/DamSqlDbWithMultiShard/DamSqlDbWithMultiShard/MultiShardConfiguration.cs

```diff
@@ -0,0 +1,245 @@
+            }
+            set
+            {
+                _shardMapName = value;
+            }
+        }
+
+        /// <summary>
+        /// Gets the server name for the Shard Map Manager database, which contains the shard maps.
+        /// </summary>
+        public static string ShardMapManagerServerName
+        {
+            get
+            {
+                return _serverName;
+            }
+        }
+
+        /// <summary>
+        /// Gets the database name for the Shard Map Manager database, which contains the shard maps.
+        /// </summary>
+        public static string ShardMapManagerDatabaseName
+        {
+            get
+            {
+                return _shardMapManagerDatabaseName;
+            }
+            set
+            {
+                _shardMapManagerDatabaseName = value;
+            }
+        }
+
+        /// <summary>
+        /// Gets the edition to use for Shards and Shard Map Manager Database if the server is an Azure SQL DB server. 
+        /// If the server is a regular SQL Server then this is ignored.
+        /// </summary>
+        public static string DatabaseEdition
+        {
+            get
+            {
+                return _databaseEdition;
+            }
+        }
+
+        #endregion
+
+        #region Shard map helper methods
+        
+        /// <summary>
+        /// Gets the shard map, if it exists. If it doesn't exist, returns null.
+        /// </summary>
+        /// <returns>shardMap</returns>
+        public static RangeShardMap<int> TryGetShardMap()
+        {
+            if (objShardMapManager == null)
+            {
+                return null;
+            }
+
+            RangeShardMap<int> shardMap;
+            bool mapExists = objShardMapManager.TryGetRangeShardMap(ShardMapName, out shardMap);
+
+            if (!mapExists)
+            {
+                return null;
+            }
+
+            return shardMap;
+        }
+
+        #endregion
+
+        #region connectionstring methods
+
+        /// <summary>
+        /// To get the connection string based on servername,shardmap databasename that are defined in the configuration file
+        /// </summary>
+        /// <returns>Connection strings that are defined in the configuration file</returns>
+        public static string GetConnectionString()
+        {
+            SqlConnectionStringBuilder sbConnStr = new SqlConnectionStringBuilder(connStr);
+            sbConnStr.DataSource = ShardMapManagerServerName;
+            sbConnStr.InitialCatalog = ShardMapManagerDatabaseName;
+            return sbConnStr.ToString();
+        }
+
+        /// <summary>
+        /// To get the connection string based on servername,master databasename that are defined in the configuration file
+        /// </summary>
+        /// <returns>Connection strings that are defined in the configuration file</returns>
+        public static string GetConnectionStringByMasterDatabase()
```

Comment:
@daisukenishino 

Similarly, please review the name of GetConnectionString**By**SelectedDatabase method.